processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -53,12 +55,17 @@ func NewDetector(_ component.ProcessorCreateSettings, dcfg internal.DetectorConf
 	if err != nil {
 		return nil, err
 	}
-	return &Detector{metadataProvider: newMetadataClient(sess), tagKeyRegexes: tagKeyRegexes}, nil
+	return &Detector{
+		metadataProvider: newMetadataClient(sess),
+		tagKeyRegexes:    tagKeyRegexes,
+		logger:           set.Logger,
+	}, nil
 }
 
 func (d *Detector) Detect(ctx context.Context) (resource pdata.Resource, schemaURL string, err error) {
 	res := pdata.NewResource()
-	if !d.metadataProvider.available(ctx) {
+	if err = d.metadataProvider.available(ctx); err != nil {
```

Comment:
Instead of doing the availability check and getting the instance ID, doesn't just trying to get the hostname perform the same check? Could we just use that instead?
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/cda651112323ecad701cb214aae2a68a3f9df91b/processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go#L70-L73

Reply:
You could use the hostname check here instead. If we can't get the hostname, then we log the debug message and return no error.

Reply:
It looks like hostname was introduced last year (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html), so it might still make sense to check for instance-id, which is guaranteed to exist.